From 9a5a4592a0eece424c33de5c9174982a4f38ace2 Mon Sep 17 00:00:00 2001
From: Yaroslav Halchenko <debian@onerussian.com>
Date: Mon, 1 Apr 2024 12:16:35 -0400
Subject: [PATCH 1/3] Add listconfigs to the example git-annex-remote-directory

This is actually a patch by  Ilya Shlyakhter <ilya_shl@alum.mit.edu>
which was shipped in conda forge package for a while but seems never got here.

Also related:
- https://git-annex.branchable.com/forum/How_to_register_arguments_for_an_external_special_remote__63__/
---
 examples/git-annex-remote-directory | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/examples/git-annex-remote-directory b/examples/git-annex-remote-directory
index c423029..c6795fb 100755
--- a/examples/git-annex-remote-directory
+++ b/examples/git-annex-remote-directory
@@ -23,6 +23,9 @@ class DirectoryRemote(ExportRemote):
             raise RemoteError("You need to set directory=")
         self._mkdir(self.directory)
 
+    def listconfigs(self):
+        return {"directory": "directory where data is stored"}
+
     def prepare(self):
         self.directory = self.annex.getconfig("directory")
         self.info = {"directory": self.directory}
-- 
2.43.0

